src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -0,0 +1,657 @@
+            }
+
+            $this->authorizationChecker = $this->container->get('security.authorization_checker');
+        }
+
+        return $this->authorizationChecker->isGranted($attributes, $object);
+    }
+
+    /**
+     * Throws an exception unless the attributes are granted against the current authentication token and optionally
+     * supplied object.
+     *
+     * @param mixed  $attributes The attributes
+     * @param mixed  $object     The object
+     * @param string $message    The message passed to the exception
+     *
+     * @throws AccessDeniedException
+     */
+    protected function denyAccessUnlessGranted($attributes, $object = null, $message = 'Access Denied.')
+    {
+        if (!$this->isGranted($attributes, $object)) {
+            throw $this->createAccessDeniedException($message);
+        }
+    }
+
+    /**
+     * Returns a rendered view.
+     *
+     * @param string $view       The view name
+     * @param array  $parameters An array of parameters to pass to the view
+     *
+     * @return string The rendered view
+     */
+    protected function renderView($view, array $parameters = array())
+    {
+        $this->populateTemplating();
+
+        if (null !== $this->templating) {
+            return $this->templating->render($view, $parameters);
+        }
+
+        $this->populateTwig('renderView');
+
+        return $this->twig->render($view, $parameters);
+    }
+
+    /**
+     * Renders a view.
+     *
+     * @param string   $view       The view name
+     * @param array    $parameters An array of parameters to pass to the view
+     * @param Response $response   A response instance
+     *
+     * @return Response A Response instance
+     */
+    protected function render($view, array $parameters = array(), Response $response = null)
+    {
+        $this->populateTemplating();
+
+        if (null !== $this->templating) {
+            return $this->templating->renderResponse($view, $parameters, $response);
+        }
+
+        $this->populateTwig('render');
+
+        if (null === $response) {
+            $response = new Response();
+        }
+
+        return $response->setContent($this->twig->render($view, $parameters));
+    }
+
+    /**
+     * Streams a view.
+     *
+     * @param string           $view       The view name
+     * @param array            $parameters An array of parameters to pass to the view
+     * @param StreamedResponse $response   A response instance
+     *
+     * @return StreamedResponse A StreamedResponse instance
+     */
+    protected function stream($view, array $parameters = array(), StreamedResponse $response = null)
+    {
+        $this->populateTemplating();
+
+        if (null !== $this->templating) {
+            $templating = $this->templating;
+
+            $callback = function () use ($templating, $view, $parameters) {
+                $templating->stream($view, $parameters);
+            };
+        } else {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method stream uses an else expression. Else is never necessary and you can simplify the code to work without else.](https://www.codacy.com/app/git_9/symfony/file/2331978825/issues/source?bid=3207821&fileBranchId=3207858#l502)